src/sqlfluff/core/rules/base.py

```diff
@@ -416,6 +416,48 @@ def raw_segments(self):
         )
 
 
+class CrawlBehavior:
```

Comment:
Extracted segment traversal code from `BaseRule.crawl()` into a separate class. This makes it easier to use different iteration strategies.